sdk/servicebus/service-bus/src/core/receiverHelper.ts

```diff
@@ -11,37 +13,61 @@ import { receiverLogger as logger } from "../log";
  * @internal
  */
 export class ReceiverHelper {
-  private _isSuspended: boolean = false;
+  private _isSuspended: boolean = true;
 
   constructor(
     private _getCurrentReceiver: () => { receiver: Receiver | undefined; logPrefix: string }
   ) {}
 
+  private _getCurrentReceiverOrError():
+    | "is undefined"
+    | "is not open"
+    | "is suspended"
+    | { receiver: Receiver | undefined; logPrefix: string } {
+    const currentReceiverData = this._getCurrentReceiver();
+
+    if (currentReceiverData.receiver == null) {
+      return "is undefined";
+    }
+
+    if (!currentReceiverData.receiver.isOpen()) {
+      return "is not open";
+    }
+
+    if (this._isSuspended) {
+      return "is suspended";
+    }
+
+    return currentReceiverData;
+  }
+
   /**
    * Adds credits to the receiver, respecting any state that
    * indicates the receiver is closed or should not continue
    * to receive more messages.
    *
    * @param credits - Number of credits to add.
-   * @returns true if credits were added, false if there is no current receiver instance
    * or `stopReceivingMessages` has been called.
    */
-  addCredit(credits: number): boolean {
-    const { receiver, logPrefix } = this._getCurrentReceiver();
+  addCredit(credits: number): void {
+    const currentReceiverOrError = this._getCurrentReceiverOrError();
 
-    if (!this.canReceiveMessages()) {
-      logger.verbose(
-        `${logPrefix} Asked to add ${credits} credits but the receiver is not able to receive messages`
-      );
-      return false;
-    }
+    if (typeof currentReceiverOrError === "string") {
+      const errorMessage = `Can't add credits to the receiver since it ${currentReceiverOrError}.`;
```

Comment:
Nit: 

Not a blocker for this PR

There is a chance that this error bubbles up to the user for whom the concept of "adding credits" may not mean anything. How about saying "request messages" instead?


```suggestion
      const errorMessage = `Cannot request messages on the receiver since it ${currentReceiverOrError}.`;
```